core-rust/core-api-server/src/core_api/conversions/receipt.rs

```diff
@@ -249,10 +250,9 @@ pub fn to_api_next_epoch(
 #[tracing::instrument(skip_all)]
 pub fn to_api_event(
     context: &MappingContext,
-    type_id: EventTypeIdentifier,
-    data: Vec<u8>,
+    event: ApplicationEvent,
 ) -> Result<models::Event, MappingError> {
-    let EventTypeIdentifier(emitter, local_type_index) = type_id;
+    let EventTypeIdentifier(emitter, local_type_index) = event.type_id;
```

Comment:
In these methods, I suggest that we use a struct-destructuring. This way, if the upstream models change / get new fields, we know we need to map them at compile time :)